lib/atlas/csv_document.rb

```diff
@@ -48,13 +48,53 @@ def initialize(path, normalizer = KEY_NORMALIZER)
       @table = CSV.table(@path.to_s, {
         converters: [YEAR_NORMALIZER, :all],
         header_converters: [normalizer],
+        # Needed to retrieve the headers in case
+        # of an otherwise empty csv file
+        return_headers: true
       })
 
-      @headers = @table.headers
+      @headers = table.headers
+
+      # Delete the header row for the internal representation -
+      # will be dynamically (re-)created when outputting
+      table.delete(0)
```

Comment:
I wrote comments in the source code for this. Do you have any suggestions how to make them clearer?